### PR TITLE
feat(gatsby-theme-bodiless): gatsby link

### LINKED ISSUE #747: Bodiless Gatsby Link
## User Story

- AS A site builder
- I WANT to a stylable gatsby link (https://www.gatsbyjs.com/docs/gatsby-link/) 
- SO THAT I benefit from the performance of having page pre-load of these links.

### Acceptance Criteria

Provide an API that checks if its relative link, it is converted to gatsby link that is stylable (that passes class to underlying a href tag.) 
This API is applied to links on test site in the following scope: tout links (/touts/), rte links (/richtext/), linked list (/list2/ page), product touts (/products page) 

### Implementation Notes

If we determine this lead to unexpected or different behavior we descope those types of links from using API and address separately.
This API willl not be applied to starter site.

### PR DESCRIPTION
**Pull request by beliayeu**
## Changes
* provides an API that checks if its relative link, it is converted to gatsby link
* applied the API to links on test site in the following scope: tout links (/touts/), rte links (/richtext/), linked list (/list2/ page), product touts (/products page)

## Related Issues

Closes #747

### CODE REVIEW

File: examples/test-site/src/components/Tout/index.tsx [hunk 1]
```diff
@@ -37,15 +38,19 @@ import {
 export const withToutEditors = flow(
   withDesign({
     Image: asEditableImage('image'),
-    ImageLink: withSidecarNodes(
-      asEditableLink('link'),
+    ImageLink: flow(
+      withSidecarNodes(
+        asEditableLink('link'),
```
Comment (dewen) [inline examples/test-site/src/components/Tout/index.tsx:43]: ```asEditableLink('link')``` => ```asEditableLink```?

File: packages/gatsby-theme-bodiless/src/dist/GatsbyLink/asGatsbyLink.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright © 2020 Johnson & Johnson
```
Comment (dewen) [inline packages/gatsby-theme-bodiless/src/dist/GatsbyLink/asGatsbyLink.tsx:2]: 2021

File: examples/test-site/src/components/Tout/index.tsx (hunk 1 above)
Comment (beliayeu) [reply examples/test-site/src/components/Tout/index.tsx:43]: I think `asEditableLink('link')` is correct
withToutEditors provides tout schema, we are injected node keys for each tout design element

File: packages/gatsby-theme-bodiless/src/dist/GatsbyLink/asGatsbyLink.tsx (hunk 2 above)
Comment (beliayeu) [reply packages/gatsby-theme-bodiless/src/dist/GatsbyLink/asGatsbyLink.tsx]: fixed in ae30ec22ca71ea3c066bb5ed269e1ada8bc28dad

**Review by dewen — APPROVED** [review]

File: packages/gatsby-theme-bodiless/src/dist/GatsbyLink/asGatsbyLink.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+type Components = {
+  GatsbyLink: CT<GatsbyLinkProps<any>>,
+  Link: CT<BodilessLinkProps>,
+};
+
+/**
+ * leveraging logic form gatsby
+ */
+const isLocalLink = (path: string) => path
```
Comment (hvanyo) [inline packages/gatsby-theme-bodiless/src/dist/GatsbyLink/asGatsbyLink.tsx:37]: @beliayeu  
Chris put in similar logic into href normalization -> https://github.com/johnsonandjohnson/Bodiless-JS/blob/dd85931f7d118cc9849b58bdcb80a27c612ea64e/packages/bodiless-components/src/Link/NormalHref.ts#L94  

Can or should any of this be reused?

cc; @wodenx

File: examples/test-site/src/components/Layout/logo.tsx [hunk 4]
```diff
@@ -36,7 +36,7 @@ export type Props = DesignableComponentsProps<LogoComponents> & HTMLProps<HTMLEl
 const logoComponents:LogoComponents = {
   SiteReturn: Div,
   SiteLogo: Img,
-  SiteLink: Link,
+  SiteLink: GatsbyLink,
```
Comment (hvanyo) [inline examples/test-site/src/components/Layout/logo.tsx:39]: Make SiteLink
      asEditableLink('logolink'),
      startWith(GatsbyLink),

Comment (beliayeu) [reply examples/test-site/src/components/Layout/logo.tsx:42]: fixed in eb8048434cee00951a60248bdfab04549c4c0f5c

**Review by dewen — APPROVED** [review]

**Review by hvanyo — APPROVED** [review]

**Pull request merged**